### ISSUE TITLE
failed to import "y-partykit/react" |- ReferenceError: WebSocket is not defined

### ISSUE THREAD

**Issue by alindsay55661** [partykit/partykit#698]
Hi,

I'm using Vite + Remix. I started with the non-yjs `PartySocket` client and everything worked fine. No issues with SSR, successful connection using multiple browsers. Then I added the `y-partykit` `onConnect` method to the server and switched out `PartySocket` with the `useYProvider` hook from `y-partykit/react` on the client as per the docs here (https://docs.partykit.io/reference/y-partykit-api/). Doing so causes the vite build to fail with the following error:

```
[vite] Error when evaluating SSR module /app/routes/_index+/whiteboard.tsx: failed to import "y-partykit/react"
|- ReferenceError: WebSocket is not defined
```

The same error occurs when importing from `y-partykit/provider` as well. I tried several vite configuration changes that others with similar (but not identical) symptoms from other projects suggested but nothing worked. Something is different between `PartySocket` and `y-partykit/react`/`y-partykit/provider`. The former imports without issue in SSR environments while the later fails.

**Comment by threepointone** [comment]
Hmm, looks like we have code paths in y-partykit that require WebSocket (or a polyfill) when initialising, and that's probably breaking in node.js environments (which I assume you're using). I can refactor that, but it might take a few days to do so cleanly. A workaround for you would be to figure out how to import and render the component that uses the `useYProvider` hook only on the client (which shouldn't change the capability of your app since it only connects on the browser/inside an effect).

**Comment by threepointone** [comment]
I just made a release with a potential fix (https://github.com/partykit/partykit/pull/699), could you try installing `y-partykit@latest` and letting me know if that fixes it for you? Thanks!

**Comment by threepointone** [comment]
closing this because I think it's fixed now, but let me know if you still have problems

**Comment by alindsay55661** [comment]
I can confirm that this update does solve the issue, thanks for the quick response!

**Comment by ozitrance** [comment]
Hi There
I am still getting this error with a fresh installation of Remix and then running commands:
`npx partykit@latest init`
`npm install yjs y-partykit`
Then changing the contents in the generated `party\index.ts` file according to instructions from: [https://docs.partykit.io/reference/y-partykit-api/](https://docs.partykit.io/reference/y-partykit-api/) and trying to use the React code w/ Remix

Any ideas...?

**Comment by threepointone** [comment]
Can you make a small repro as a git repo? Happy to have a look.

**Comment by ozitrance** [comment]
Thanks a lot for the super quick response. I'm still learning React, so it could easily be just something I missed but I followed the instructions and was able to run Partykit without Yjs without any problems so I'm not sure why it won't work with Yjs.

Anyways I uploaded the simple setup to [`https://github.com/ozitrance/party-yjs`](https://github.com/ozitrance/party-yjs)
Let me know if you need anything else

**Comment by threepointone** [comment]
Fixed and released in `y-partykit@0.0.17`, sorry for the trouble!

**Comment by ozitrance** [comment]
Working now, no errors :) Thank you so much!
